autorest/codegen/__init__.py

```diff
@@ -134,7 +149,7 @@ def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union
         code_model.setup_client_input_parameters(yaml_data)
 
         # Get my namespace
-        namespace = self._autorestapi.get_value("namespace")
+        namespace = code_model.options.get("namespace", self._autorestapi.get_value("namespace"))
```

Comment:
@iscai-msft why would there be a namesapce in the code_model.options?

Reply:
yeah you're right there isn't any, @msyyc I don't think we need to change the code here

Reply:
`namesapce` in the `code_model.options` is needed to render template for `package-mode`. Please see here: https://github.com/Azure/autorest.python/pull/1112/files#r794987407